sys/ztimer/periph_timer.c

```diff
@@ -104,12 +106,16 @@ static const ztimer_ops_t _ztimer_periph_timer_ops = {
 #endif
 };
 
-void ztimer_periph_timer_init(ztimer_periph_timer_t *clock, tim_t dev,
-                              uint32_t freq, uint32_t max_val)
+uint32_t ztimer_periph_timer_init(ztimer_periph_timer_t *clock, tim_t dev,
+                                  uint32_t freq, uint32_t max_val)
 {
     clock->dev = dev;
     clock->super.ops = &_ztimer_periph_timer_ops;
     clock->super.max_value = max_val;
+
+    if (IS_USED(MODULE_PERIPH_TIMER_QUERY_FREQS)) {
+        freq = timer_get_closest_freq(dev, freq);
```

Comment:
Should we print a warning if the difference is too big or can ztimer deal with arbitrary frequencies?

Reply:
It should do exactly that (with `DEVELHELP=1`) later on, unless `ztimer_convert_frac` is used, which just does proper frequency conversion.